services/streammanager/kafka/kafkamanager.go

```diff
@@ -418,11 +467,26 @@ func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timest
 			pkgLogger.Errorf("batch from topic %s is missing the userId attribute", topic)
 			continue
 		}
-
 		marshalledMsg, err := json.Marshal(message)
 		if err != nil {
 			return nil, err
 		}
+		codecs := p.getCodecs()
+		if len(codecs) > 0 {
+			schemaId, _ := data["schemaId"].(string)
+			if schemaId == "" {
+				return nil, fmt.Errorf("schemaId is not available for the event of index:%d", i)
+			}
+			if codec, ok := codecs[schemaId]; ok {
+				marshalledMsg, err = serialize(marshalledMsg, codec)
+				if err != nil {
+					return nil, fmt.Errorf("unable to serialize the event of index: %d, with error: %s", i, err)
+				}
+			} else {
+				return nil, fmt.Errorf("unable to find schema with schemaId: %v", schemaId)
+			}
```

Comment:
This can also be simplified by removing the `if/else`. It's doable with a single `if`.

Reply:
Updated.